pkg/scheduler/factory_test.go

```diff
@@ -426,16 +346,70 @@ func TestDefaultErrorFunc(t *testing.T) {
 			if diff := cmp.Diff(tt.expect, got); diff != "" {
 				t.Errorf("Unexpected pod (-want, +got): %s", diff)
 			}
+		})
+	}
+}
 
-			if !tt.testBackoff {
-				return
+func TestDefaultErrorFunc_NodeNotFound(t *testing.T) {
+	nodeFoo := &v1.Node{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}
+	nodeBar := &v1.Node{ObjectMeta: metav1.ObjectMeta{Name: "bar"}}
+	testPod := &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-pod", Namespace: "default"}}
+	tests := []struct {
+		name             string
+		nodes            []v1.Node
+		nodeNameToDelete string
+		injectErr        error
+		expectNodeNames  sets.String
+	}{
+		{
+			name:             "node is deleted during a scheduling cycle",
+			nodes:            []v1.Node{*nodeFoo, *nodeBar},
+			nodeNameToDelete: "foo",
+			injectErr:        apierrors.NewNotFound(apicore.Resource("node"), nodeFoo.Name),
+			expectNodeNames:  sets.NewString("bar"),
+		},
+		{
+			name:            "node is not deleted but NodeNotFound is received incorrectly",
```

Comment:
Why are we testing this?

Reply:
to cover the path "err == nil" here, so **don't** remove the Node from cache:

https://github.com/kubernetes/kubernetes/blob/438debfb7263619d444c2eb9f6b52747415a9f24/pkg/scheduler/factory.go#L464-L470

Reply:
sg